### PR TITLE
feat(order): add magento 2.4.1 driver models and transforms

### PR DESCRIPTION
**Pull request by griest024**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Fixes: N/A

## What is the new behavior?
- Adds official Magento models
- Adds a service for generating Daffodil and Magento mock data with matching info
- Adds order transforms

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

## Other information

### CODE REVIEW

File: libs/order/driver/magento/2.4.1/src/transforms/responses/order.ts [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+  return {
+    attribute_label: option.label,
+	  value_label: option.value
+  }
+}
+
+function transformBundleOption(option: MagentoOrderBundleItemSelectedOption): DaffCompositeOrderItemOption {
+  return {
+    option_label: option.label,
+	  value_label: option.values[0].product_name
```
Comment (lderrickable) [inline libs/order/driver/magento/2.4.1/src/transforms/responses/order.ts:87]: Needs a null check?

Comment (griest024) [reply libs/order/driver/magento/2.4.1/src/transforms/responses/order.ts]: I haven't seen that `values` can ever be `null` but the schema doesn't prevent it so I'll add it to be safe. added

**Review by lderrickable — DISMISSED** [review]

**Comment by lderrickable** [comment]
you have unverified commits, which is preventing the merge

**Comment by griest024** [comment]
@lderrickable oh yeah, I guess my windows git instance isn't configed correctly, I'll fix

**Comment by griest024** [comment]
@lderrickable okay got it configed

**Review by lderrickable — APPROVED** [review]

**Pull request merged**